shell/common/run_configuration.h

```diff
@@ -18,38 +18,162 @@
 
 namespace flutter {
 
-/// Configuration pertaining to how we will execute the Dart code.
+//------------------------------------------------------------------------------
+/// @brief      Specifies all the configuration required by the runtime library
+///             to launch the root isolate. This object may be created on any
+///             thread but must be given to the |Run| call of the |Engine| on
+///             the UI thread. The configuration object is used to specify how
+///             the root isolate finds its snapshots, assets, root library and
+///             the "main" entrypoint.
 ///
-/// For example: what will be the first function we execute (entrypoint).
 class RunConfiguration {
  public:
+  //----------------------------------------------------------------------------
+  /// @brief      Attempt to infer a run configuration from the settings object.
+  ///             This tries to create a run configuration with sensible
+  ///             defaults for the given Dart VM runtime mode. In JIT mode, this
+  ///             will attempt to look for the VM and isolate snapshots in the
+  ///             assets directory (must be specified in settings). In AOT mode,
+  ///             it will attempt to look for known snapshot symbols in the
+  ///             currently currently loaded process. The entrypoint default to
+  ///             the "main" method in the root library.
+  ///
+  /// @param[in]  settings   The settings object used to look for the various
+  ///                        snapshots and settings. This is usually initialized
+  ///                        from command line arguments.
+  /// @param[in]  io_worker  An optional IO worker. Resolving and reading the
+  ///                        various snapshots may be slow. Providing an IO
+  ///                        worker will ensure that realization of these
+  ///                        snapshots happens on a worker thread instead of the
+  ///                        calling thread. Note that the work done to realize
+  ///                        the snapshots may occur after this call returns. It
+  ///                        is embedder responsibility to make sure the thread
+  ///                        for the serial worker is kept alive for as long as
+  ///                        the shell of the engine this run configuration is
+  ///                        given to is alive.
```

Comment:
I wonder if this would scan a little easier:

.. for the serial worker is kept alive for the lifetime of the shell associated with the engine that this run configuration is given to.

Reply:
Done. It is still a bit hard to follow but accurate.